src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogger.cs

```diff
@@ -123,6 +125,23 @@ private void StartFunction(FunctionInstanceLogEntry item)
             }
         }
 
+        internal object LogInvocationMetrics(FunctionMetadata metadata)
+        {
+            // log events for each of the binding types used
+            foreach (var binding in metadata.Bindings)
+            {
+                string eventName = _bindingMetricEventNames.GetOrAdd(binding, (existing) =>
```

Comment:
Caching these metric event names on this hotpath. No need to reformat these on each invocation for each binding.